### PR TITLE
Attempt to make spec/helper.js more robust.

### PR DESCRIPTION
**Pull request by rolandpoulter**

### CODE REVIEW

File: spec/helper.js [hunk 1]
```diff
@@ -217,11 +217,12 @@ global.helper = {
      * @returns {Promise}
      */
     stop: function () {
-        if (this.core) {
-            return this.core.stop();
-        } else {
-            return bluebird.resolve();
-        }
+      var promises = [new Promise(function (resolve) {
+        setTimeout(resolve);
```
Comment (benbp) [inline spec/helper.js]: Why is this needed?

File: spec/helper.js [hunk 2]
```diff
@@ -217,11 +217,12 @@ global.helper = {
...
+      })];
+      if (this.core) promises.push(this.core.stop());
+      promises.push(bluebird.resolve());
+      Promise.all(promises);
```
Comment (benbp) [inline spec/helper.js]: Need to return.

File: spec/helper.js [hunk 3]
```diff
@@ -217,11 +217,12 @@ global.helper = {
...
-        if (this.core) {
-            return this.core.stop();
-        } else {
-            return bluebird.resolve();
-        }
+      var promises = [new Promise(function (resolve) {
+        setTimeout(resolve);
+      })];
+      if (this.core) promises.push(this.core.stop());
+      promises.push(bluebird.resolve());
```
Comment (benbp) [inline spec/helper.js]: Can just do Promise.resolve() instead since you already have it in scope.

File: spec/helper.js [hunk 4]
```diff
@@ -217,11 +217,12 @@ global.helper = {
...
     stop: function () {
-        if (this.core) {
-            return this.core.stop();
-        } else {
-            return bluebird.resolve();
-        }
+      var promises = [new Promise(function (resolve) {
+        setTimeout(resolve);
+      })];
+      if (this.core) promises.push(this.core.stop());
```
Comment (benbp) [inline spec/helper.js]: I think we shy away from one line ifs in our general, unwritten style conventions.

**Comment by jfrey** [comment]
Need a description.

File: spec/helper.js (hunk 1 above)
Comment (rolandpoulter) [inline spec/helper.js]: I was hoping that having a timeout between tests would yield to `nextTick` and `setImmediate` to be called. Incase it is causing race conditions.

**Comment by rolandpoulter** [comment]
Closing for now, I meant to open this PR against my fork.

**Comment by benbp** [comment]
lol
